File: hw/pastor/u1/hw16Libraries/Python/calculator_income_tax.py
import income_tax_lib as ir


def _reed_float(message: str) -> float:
    while True:
        try:
            return float(input(message))
        except ValueError:
            print("Incorrect data, please enter a number (e.g 1254.66)")

def _reed_int(message: str) -> int:
    while True:
        try:
            return float(input(message))
        except ValueError:
            print("Incorrect data, please enter a number (e.g 16)")

def main():
    print("----- WELCOME TO THE INCOME TAX ECUADOR 2025 BY MATHEWS PASTOR -----")

    mounthy_salary = _reed_float("Enter your average monthly salary: ")
    annual_personal_expenses = _reed_float("Enter your total Annual Personal Expenses: ")
    depents = _reed_int("Enter Number of Dependents: ")
    annual_withholding = _reed_float("Enter Annual Withholdings (already paid): ")
    
    annual_gross_income = mounthy_salary * 12
    annual_iees_contributions = annual_gross_income * 0.0945

    balance = ir._final_balance(annual_gross_income, annual_iees_contributions, annual_personal_expenses, depents, annual_withholding)
    if balance > 0:
        print(f"RESULT: Tax Payable: ${balance}")
    else:
        print(f"RESULT: Balance in favor: ${abs(balance)}")
    
main()

File: hw/pastor/u1/hw16Libraries/Javascript/calculatorIncomeTax.js
import {finalBalance} from './incomeTaxlib.js';
import * as readline from 'node:readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const askQuestion = (query) => {
    return new Promise((resolve) => {
        rl.question(query, resolve);
    });
};

const readFloat = async (message) => {
    while (true) {
        const input = await askQuestion(message);
        const value = parseFloat(input);
        if (!isNaN(value)) {
            return value;
        }
        console.log("Incorrect data, please enter a number (e.g 1254.66)");
    }
};

const readInt = async (message) => {
    while (true) {
        const input = await askQuestion(message);
        const value = parseInt(input, 10);
        if (!isNaN(value) && parseFloat(input) === value) {
            return value;
        }
        console.log("Incorrect data, please enter a number (e.g 16)");
    }
};

async function main() {
    console.log("----- WELCOME TO THE INCOME TAX ECUADOR 2025 BY MATHEWS PASTOR -----");

    const monthlySalary = await readFloat("Enter your average monthly salary: ");
    const annualPersonalExpenses = await readFloat("Enter your total Annual Personal Expenses: ");
    const depents = await readInt("Enter Number of Dependents: ");
    const annualWithholding = await readFloat("Enter Annual Withholdings (already paid): ");
    rl.close();

    const annualGrossIncome = monthlySalary * 12;
    const annualIeesContributions = annualGrossIncome * 0.0945;

    const balance = finalBalance(
        annualGrossIncome,
        annualIeesContributions,
        annualPersonalExpenses,
        depents,
        annualWithholding
    );

    if (balance > 0) {
        console.log(`\nRESULT: Tax Payable: $${balance.toFixed(2)}`);
    } else {
        console.log(`\nRESULT: Balance in favor: $${Math.abs(balance).toFixed(2)}`);
    }
}

main();